arch/Kconfig

```
@@ -864,6 +864,17 @@ config CODE_DATA_RELOCATION
 	  the target regions should be specified in CMakeLists.txt using
 	  zephyr_code_relocate().
 
+config DELAY_DATA_RELOCATION
+	bool "Support delayed data relocation/bss init"
+	depends on ARCH_HAS_CODE_DATA_RELOCATION
+	depends on CODE_DATA_RELOCATION
```

`CODE_DATA_RELOCATION` already depends on `ARCH_HAS_CODE_DATA_RELOCATION`, no need to add it again.

```suggestion
	depends on CODE_DATA_RELOCATION
```